File: PythonVersion/cell.py
class Automata:
    def __init__(self, x=0, y=0, alive=False):
        self._x = x
        self._y = y
        self._alive = alive

    def __str__(self):
        if self._alive == True:
            return " O "
        return " * "

    def set_alive(self):
        self._alive = True

    def set_dead(self):
        self._alive = False

    def mortality_status(self):
        return self._alive

    def set_coordinates(self, x, y):
        self._x = x
        self._y = y

    def coordinates(self):
        return (self._x, self._y)

File: rustversion/src/cell.rs
#[derive(Debug, Clone, Copy)]
pub struct Cell {
    x: i32,
    y: i32,
    alive: bool,
}

pub fn get_new_cell() -> Cell {
    return Cell {
        x: 0,
        y: 0,
        alive: false,
    };
}

impl Cell {
    pub fn show(&self) {
        if self.alive {
            print!(" O ");
        } else {
            print!(" * ");
        }
    }

    pub fn set_alive(&mut self) {
        self.alive = true;
    }

    pub fn set_dead(&mut self) {
        self.alive = false;
    }

    pub fn is_alive(&self) -> bool {
        return self.alive == true;
    }
    pub fn is_dead(&self) -> bool {
        return self.alive == false;
    }

    // pub fn set_coordinates(&mut self, x: i32, y: i32) {
    //     self.x = x;
    //     self.y = y;
    // }

    pub fn get_coordinates(&self) -> (i32, i32) {
        return (self.x, self.y);
    }
}
